lib/mpl_toolkits/axisartist/axis_artist.py

```diff
@@ -152,13 +154,61 @@ def get_markeredgecolor(self):
     def get_markeredgewidth(self):
         return self.get_attribute_from_ref_artist("markeredgewidth")
 
+    def set_tickdir(self, tickdir):
+        _api.check_in_list(self._tick_paths, tickdir=tickdir)
+        self._tickdir = tickdir
+
+    def get_tickdir(self):
+        return self._tickdir
+
+    def set_tick_orientation(self, mode):
+        """
+        Set how tick orientation will be determined.
+
+        Parameters
+        ----------
+        mode : {"parallel", "normal", "auto"}
+          'parallel' - ticks along the grid lines
+          'normal' - ticks normal to axis line.
+          'auto' - 'normal' if tickdir is 'out' else 'parallel'
+        """
+        _api.check_in_list(["auto", "normal", "parallel"],
+                           tick_orientation=mode)
+        self._tick_orientation = mode
+
+    def get_tick_orientation(self, interpret_auto=True):
+        """
+        Return orientation of ticks.
+
+        Parameters
+        ----------
+        interpret_auto : bool, default if True
+          If True and tick_orientation is 'auto', return the
+          interpreted value ('normal' if dicktir is out, else 'parallel).
+          If False, return 'auto' as 'auto'.
+
+        Returns
+        -------
+        tick_orientation : {"parallel", "normal", "auto"}
+        """
+        tick_orientation = self._tick_orientation
+        if tick_orientation == "auto" and interpret_auto:
+            tick_orientation = ("normal" if self._tickdir == "out"
+                                else "parallel")
+
+        return tick_orientation
+
     def set_tick_out(self, b):
```

Comment:
Maybe we should deprecate these now?

Reply:
At what version do we want it to be deprecated?

Reply:
I'd say 3.7.

Reply:
I think this could fit in our normal warn for one release, remove the next pattern, but would defer to you if you wanted to put the new version in for a cycle before we started (or pushed the removal date out further).